### PR TITLE
Object tree component

### PR DESCRIPTION
**Pull request by dottorblaster**
# Description
New ObjectTree component needed for the checks UI

2023-03-02-090531_214x152_scrot

## How was this tested?

Jest tests

### CODE REVIEW

File: assets/js/components/ObjectTree/ObjectTree.test.jsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+    expect(count).toBe(children.length - 1);
+
+    const [
+      {
+        children: [firstChildId, _second, _third, fourthChildId],
+      },
+    ] = children;
+
+    expect(children[0].id).toBe(0);
```
Comment (arbulu89) [inline assets/js/components/ObjectTree/ObjectTree.test.jsx:45]: Damn, this is hard to understand hehe

**Review by arbulu89 — APPROVED** [review]
Super nice @dottorblaster 
It looks great.
Could you add an updated screenshot with the expanded array? :stuck_out_tongue_closed_eyes: 
To see it full capacity hehe

Maybe in the future we will need to add some modification to don't show the whole tree in the results page, only a partial part and expandable with `...` or something. Otherwise big trees will become too big

PD:
As a reference for others. The code is based on: https://www.npmjs.com/package/react-accessible-treeview

**Comment by dottorblaster** [comment]
@arbulu89 since an array has an `Object` prototype it actually works as an object :grimacing: 

image

@jagabomb did you have some idea about representing arrays visually?

**Comment by jagabomb** [comment]
@dottorblaster So I haven't really considered the Arrays being commonly represented. But based on the screenshot attached I think this does the job. We can always improve it if there is any concerns from with the users.

File: assets/js/components/ObjectTree/tree.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    id,
+    name: tree.name,
+    children: childrenIds,
+    parent,
+  };
+
+  return { count, children: [node, ...children] };
+};
+
+export const treeify = (data) => ({
```
[the remark this answers is not in the document]
Comment (dottorblaster) [reply assets/js/components/ObjectTree/tree.js:43]: I was dubious then I noticed that you were actually right :+1:

File: assets/js/components/ObjectTree/ObjectTree.test.jsx (hunk 1 above)
Comment (dottorblaster) [reply assets/js/components/ObjectTree/ObjectTree.test.jsx:45]: LOL yes it's dumb but I wanted to test that the `id` field was actually the index of the array, as I wasn't giving it for granted. I foyu want I can remove this expectation

File: assets/js/components/ObjectTree/ObjectTree.test.jsx [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+import ObjectTree from '.';
+import { flattenTree, treeify } from './tree';
+
+describe('ObjectTree component', () => {
+  it('should render correctly', () => {
+    const data = objectTreeFactory.build();
+    const {
+      number,
+      string,
```
[the remark this answers is not in the document]
Comment (dottorblaster) [reply assets/js/components/ObjectTree/ObjectTree.test.jsx:15]: Right! :+1:

**Review by nelsonkopliku — APPROVED** [review]
Cool stuff!

**Pull request merged**